### DISCUSSION TITLE
Multiple "Tokio:: spawn" using the same "TcpStream" variable

### DISCUSSION THREAD

**Discussion by yunfengwangluo** [tokio-rs/tokio#6349] [General]
I am using "Tokio:: net:: TcpStream". When I try to use the same TcpStream in different functions simultaneously, ownership issues naturally arise. I tried Arc<Mutex<TcpStream>, but doing so alone won't solve my problem. When I use the library std:: net: TcpStream or net2, there is a "try_clone" method that can be called. I checked the code of tokio:: net and found that the net used by tokio mainly relies on the "mio:: net" library. I don't know what advantages this library has compared to std:: net. But currently, I am unable to implement my requirements in multiple functions because I need to communicate with other interfaces. I want my code to implement two functions, one to read data and the other to write data.

Please forgive my expression skills, I am very sorry!

If you have free time, please help me solve my problem. Thank you very much!

**Comment by Darksonn** [comment]
I suggest one of the following solutions:

1. If you just need one reader and one writer, then use [`TcpStream::into_split`](https://docs.rs/tokio/latest/tokio/net/struct.TcpStream.html#method.into_split).
2. Use [the actor pattern](https://ryhl.io/blog/actors-with-tokio/) to avoid sharing the IO resource.
3. Stop sharing the tcp stream entirely. If you only need it in one task, then there is no problem.
